cloudinary/api_client/tcp_keep_alive_manager.py

```diff
@@ -0,0 +1,119 @@
+import socket
+import sys
+
+from urllib3 import HTTPSConnectionPool, HTTPConnectionPool, PoolManager, ProxyManager
+
+# Inspired by:
+# https://github.com/finbourne/lusid-sdk-python/blob/b813882e4f1777ea78670a03a7596486639e6f40/sdk/lusid/tcp/tcp_keep_alive_probes.py
+
+# The content to send on Mac OS in the TCP Keep Alive probe
+TCP_KEEPALIVE = 0x10
+# The maximum time to keep the connection idle before sending probes
+TCP_KEEP_IDLE = 60
+# The interval between probes
+TCP_KEEPALIVE_INTERVAL = 60
+# The maximum number of failed probes before terminating the connection
+TCP_KEEP_CNT = 3
+
+
+class TCPKeepAliveValidationMethods:
+    """
+    This class contains a single method whose sole purpose is to set up TCP Keep Alive probes on the socket for a
+    connection. This is necessary for long-running requests which will be silently terminated by the AWS Network Load
```

Comment:
The testing we had done was with NAT Gateways (https://docs.aws.amazon.com/vpc/latest/userguide/nat-gateway-troubleshooting.html#nat-gateway-troubleshooting-timeout) whereas the comment says `AWS Network Load Balancer`.